### PR TITLE
Jahanzaib test branch

### PR DESCRIPTION
**Pull request by jehanxaibahmed**
Instructor to User Chat

### CODE REVIEW

File: app/Http/Controllers/ChatRequestsController.php [hunk 1]
```diff
@@ -30,66 +31,79 @@ public function index()
...
-
+        
+        $message = ConversationMessage::create([    
+            'conversation_id' => $conversation->id,
+            'sender_id' => $senderId,
+            'receiver_id' => $receiverId,
+            'body' => $message
+        ]);
+        
+        $conversation->last_time_message = now();
```
Comment (guardianstraining) [inline app/Http/Controllers/ChatRequestsController.php:78]: last_time_message is not needed

The timestamp attribute in the schema provides a created_at & updated_at column, so the updated_at column can be used in this case.

To keep the controller clean, firstOr create should replace line 53 - 69. See example below

 $conversation = Conversation::firstOrCreate(
            ['sender_id' => $senderId, 'receiver_id' => $receiverId ],
            ['chat_id' => $chatId]
        );

File: app/Http/Controllers/ChatRequestsController.php [hunk 2]
```diff
@@ -30,66 +31,79 @@ public function index()
...
             ]);
-            $chat = Chat::find($request->request_id);
-            $this->checkconversation($chat->created_by_id, $request->input('instructor_id'), $chat->chat_string);
-            $chat->assigned_to_id = $request->input('instructor_id');
-            $chat->save();
+            
+            $instructor_id = $request->input('instructor_id');
+            $status = $request->input('status');
+
+            if ($chat_request->assigned_to_id !== intval($instructor_id)) {
```
Comment (guardianstraining) [inline app/Http/Controllers/ChatRequestsController.php:98]: Any specific reason for using intval?

File: resources/views/jambasangsang/backend/chat_requests/partials/show.blade.php [hunk 3]
```diff
@@ -2,12 +2,12 @@
     <div class="card-body">
 
         <div class="col-12">
-            <h6 class="mt-3 card-title">Subject: {!! $request->issue !!} </h6>
+            <h6 class="mt-3 card-title">Subject: {!! $chat_request->issue !!} </h6>
```
Comment (guardianstraining) [inline resources/views/jambasangsang/backend/chat_requests/partials/show.blade.php:5]: Why are you using double exclamation?
Please understand the use case

File: resources/views/jambasangsang/backend/chat_requests/partials/show.blade.php [hunk 4]
```diff
@@ -16,42 +16,46 @@
...
+            <div class="col-12 mt-3">
+                <livewire:backend.select.status-select wire:key="chat_status_selector"
+                    :selected="isset($chat_request->status)
+                        ? $chat_request->status
+                        : old('status')" />
             </div>
-
             <div class="col-12 mt-3">
-                <input type="hidden" name="request_id" value="{!! $request->id !!}" />
                 <button id="payment-button" type="submit" class="btn btn-info btn-block">
```
Comment (guardianstraining) [inline resources/views/jambasangsang/backend/chat_requests/partials/show.blade.php:56]: rename payment button to match what's happening on this page

File: resources/views/jambasangsang/backend/live_chat/assign.blade.php [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+            <div class="col-12 mt-3">
+                <livewire:backend.select.status-select wire:key="chat_status_selector"
+                    :selected="isset($chat->chat->status)
+                        ? $chat->chat->status
+                        : old('status')" />
+            </div>
+
+            <div class="col-12 mt-3">
+                <button id="payment-button" type="submit" class="btn btn-info btn-block">
```
Comment (guardianstraining) [inline resources/views/jambasangsang/backend/live_chat/assign.blade.php:22]: payment button again

File: resources/views/layouts/backend/master.blade.php [hunk 6]
```diff
@@ -25,7 +25,8 @@
     <link rel="stylesheet" href="{{ asset('jambasangsang/backend/vendors/jqvmap/dist/jqvmap.min.css') }}">
 
     <link rel="stylesheet" href="{{ asset('jambasangsang/backend/assets/css/style.css') }}">
-
+    <script src="{{ asset('jambasangsang/backend/vendors/jquery/dist/jquery.min.js') }}"></script>
```
Comment (guardianstraining) [inline resources/views/layouts/backend/master.blade.php:28]: Did you add a new script? seems there's duplicate jquery.min.js on this page

**Review by guardianstraining — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/ChatRequestsController.php (hunk 1 above)
Comment (jehanxaibahmed) [reply app/Http/Controllers/ChatRequestsController.php:78]: We are using this last_time_message in the conversation table and update it on every message sent while the update_at column is updated when we do some updates in the conversation table.

**Review by guardianstraining — APPROVED** [review]

**Pull request merged**